bench/run.cpp

```diff
@@ -29,12 +29,12 @@ static void BM_decode_polygon_fixture(benchmark::State& state) // NOLINT google-
 {
     std::string buffer = open_tile("test/mvt-fixtures/fixtures/019/tile.mvt");
     vtzero::vector_tile tile(buffer);
-    auto layer = tile.next_layer();
-    auto feature = layer.next_feature();
+    auto layer = tile.begin();
+    auto feature = (*layer).begin();
```

Comment:
@joto it seems that I could not do `layer->begin()` here from the layer_iterator. Is this expected?

Reply:
Oversight by me. Missing the `operator->`. I will add that.

Reply:
Ah, no. I looked into this again and there is a good reason: Dereferencing the iterator returned by `tile.begin()` doesn't give you access to an existing layer object, it will create a layer object for you. So you always want to dereference that and use it from there, otherwise it would have to parse the layer data again and again. I can probably make the `->` work, but using it is an antipattern. You should always do something like this:

```
auto layer = *tile.begin();
auto feature = *layer.begin();
```